packages/cloudflare/src/handler.ts

```diff
@@ -0,0 +1,138 @@
+export function withSentry<E extends ExportedHandler<any>>(
+  optionsCallback: (env: ExtractEnv<E>) => Options,
+  handler: E,
+): E {
+  setAsyncLocalStorageAsyncContextStrategy();
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+  if ('fetch' in handler && typeof handler.fetch === 'function' && !(handler.fetch as any).__SENTRY_INSTRUMENTED__) {
+    handler.fetch = new Proxy(handler.fetch, {
+      apply(target, thisArg, args: Parameters<ExportedHandlerFetchHandler<ExtractEnv<E>>>) {
+        const [request, env, context] = args;
+        return withIsolationScope(isolationScope => {
+          const options = optionsCallback(env);
+          const client = init(options);
+          isolationScope.setClient(client);
+
+          const attributes: SpanAttributes = {
+            [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.http.cloudflare-worker',
+            [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'url',
+            [SEMANTIC_ATTRIBUTE_SENTRY_OP]: 'http.server',
+            ['http.request.method']: request.method,
+            ['url.full']: request.url,
+          };
+
+          const contentLength = request.headers.get('content-length');
+          if (contentLength) {
+            attributes['http.request.body.size'] = parseInt(contentLength, 10);
+          }
+
+          let pathname = '';
+          try {
+            const url = new URL(request.url);
+            pathname = url.pathname;
+            attributes['server.address'] = url.hostname;
+            attributes['url.scheme'] = url.protocol.replace(':', '');
+          } catch {
+            // skip
+          }
+
+          addRequest(isolationScope, request);
+          addCloudResourceContext(isolationScope);
+          if (request.cf) {
+            addCultureContext(isolationScope, request.cf);
+            attributes['network.protocol.name'] = request.cf.httpProtocol;
+          }
+
+          const routeName = `${request.method} ${pathname ? stripUrlQueryAndFragment(pathname) : '/'}`;
+
+          return continueTrace(
+            { sentryTrace: request.headers.get('sentry-trace') || '', baggage: request.headers.get('baggage') },
+            () => {
+              // Note: This span will not have a duration unless I/O happens in the handler. This is
+              // because of how the cloudflare workers runtime works.
+              // See: https://developers.cloudflare.com/workers/runtime-apis/performance/
+              return startSpan(
+                {
+                  name: routeName,
+                  attributes,
+                },
+                async span => {
+                  try {
+                    const res = await (target.apply(thisArg, args) as ReturnType<typeof target>);
+                    setHttpStatus(span, res.status);
+                    return res;
+                  } catch (e) {
+                    captureException(e, { mechanism: { handled: false } });
```

Comment:
q: I've seen some sdks add additional keys to mechanism, e.g. `type: 'bun'`, do we want that here too? why or why not?

Reply:
As per the [develop docs](https://develop.sentry.dev/sdk/event-payloads/types/#typedef-Mechanism)

> Required unique identifier of this mechanism determining rendering and processing of the mechanism data.

> In the Python SDK this is merely the name of the framework integration that produced the exception, while for native it is e.g. "minidump" or "applecrashreport". 

Changed with 309c05ab1163e2b7671b63845ae28697becb117a